client/my-sites/patterns/home/index.tsx

```diff
@@ -34,8 +54,54 @@ export const PatternsHomePage = ( {
 	const locale = useLocale();
 	const isLoggedIn = useSelector( isUserLoggedIn );
 
+	const [ searchTerm, setSearchTerm ] = useState( () => {
+		if ( typeof window !== 'undefined' ) {
+			const params = new URLSearchParams( window.location.search );
+			return params.get( 's' ) ?? '';
+		}
+
+		return '';
+	} );
+
+	const isPatternsDisplaying = Boolean( category || searchTerm );
+
 	const { data: categories } = usePatternCategories( locale );
-	const { data: patterns } = usePatterns( locale, category );
+	const { data: patterns = [] } = usePatterns( locale, category, {
+		enabled: isPatternsDisplaying,
+	} );
+
+	const patternSearchResults = filterPatternsByQuery( patterns, searchTerm );
```

Comment:
A small suggestion to be more consistent with the `searchTerm` parameter:
```suggestion
	const patternSearchResults = filterPatternsByTerm( patterns, searchTerm );
```